modules/apps/transfer/keeper/relay.go

```diff
@@ -129,7 +130,22 @@ func (k Keeper) sendTransfer(
 
 	packetData := types.NewFungibleTokenPacketDataV2(tokens, sender.String(), receiver, memo)
 
-	sequence, err := k.ics4Wrapper.SendPacket(ctx, channelCap, sourcePort, sourceChannel, timeoutHeight, timeoutTimestamp, packetData.GetBytes())
+	appVersion, found := k.ics4Wrapper.GetAppVersion(ctx, sourcePort, sourceChannel)
+	if !found {
+		return 0, errorsmod.Wrapf(ibcerrors.ErrNotFound, "app version not found for port %s and channel %s", sourcePort, sourceChannel)
+	}
+
+	packetDataBytes := packetData.GetBytes()
+	if appVersion == types.V1 {
+		// PacketDataV2ToV1 checks that length of tokens must be == 1.
+		packetDataV1, err := convert.PacketDataV2ToV1(packetData)
+		if err != nil {
+			return 0, err
+		}
+		packetDataBytes = packetDataV1.GetBytes()
+	}
```

Comment:
I am curious to know why you chose this approach of creating V2 fungible token packet data + converting to V1, instead of creating the right version of fungible token packet data based on the channel version. So something like this:

```go
var packetDataBytes []byte
switch appVersion {
case types.V1:
	// maybe check that tokens length == 1
	token := tokens[0]
	packetData := types.NewFungibleTokenPacketData(token.Denom.FullPath(), token.Amount, sender.String(), receiver, memo)
	packetDataBytes = packetData.GetBytes()
case types.V2:
	packetData := types.NewFungibleTokenPacketDataV2(tokens, sender.String(), receiver, memo)
	packetDataBytes = packetData.GetBytes()
default:
	// return error
}
```